src/map_extras.cpp

```diff
@@ -328,30 +327,61 @@ void mx_roadblock( map &m, const tripoint &abs_sub )
     }
     if( mil ) { //Military doesn't joke around with their barricades!
 
-        if( north.find( "road_" ) == 0 && south.find( "road_" ) == 0 ) {
-            rotated = true;
-            //Rotate the terrain -90 so that all of the items will be in the correct position
-            //when the entire map is rotated at the end
-            m.rotate( 3 );
+        if( north.find( "road_" ) == 0 ) {
```

Comment:
it might be worthwhile caching the results of the searches in some booleans to save repeating the comparisons for the turrets later.